python/cudf/cudf/utils/_numba.py

```diff
@@ -7,6 +7,19 @@
 
 from numba import config as numba_config
 
+try:
+    from pynvjitlink.patch import (
+        patch_numba_linker as patch_numba_linker_pynvjitlink,
+    )
+except ImportError:
+
+    def patch_numba_linker_pynvjitlink():
+        warnings.warn(
+            "Minor version compatibility requires pynvjitlink. "
+            "Please pip install pynvjitlink to proceed using cuDF."
```

Comment:
Depending on the timing of pynvjitlink being shipped, we'll want to do one of the following:
- If pynvjitlink isn't shipped by the time we release, we should remove this note and say something like before (explain the MVC problem and direct users to match their driver and toolkit)
- If pynvjitlink _is_ available in time, we should give the full install command, like `pip install pynvjitlink --extra-index-url=https://pypi.nvidia.com`

Reply:
How do you feel about keeping the original warning to be safe then. The changes will still allow `pynvjitlink` to work once it's out, but we won't document it or direct users to it in any official way until then.

Reply:
Yes, do that. We can change the warning during code freeze if we ship pynvjitlink in time.